addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/KM200ThingType.java

```diff
@@ -0,0 +1,356 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.km200;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+
+/**
+ * The KM200ServiceTypes enum is representing the things
+ *
+ * @author Markus Eckhardt
+ *
+ */
+public enum KM200ThingType {
+    GATEWAY {
+        @Override
```

Comment:
I don't think its wise to have so many enum specific method bodies. Maybe you can just create an interface and multiple different implementations. Maybe this is good one to have some small discussion about.

Reply:
I changed the enums and the handling. All data are now in one enum and it is extended with the contructor and default data handling. Please take a look....